src/charging_session.py

```diff
@@ -0,0 +1,92 @@
+import os
+import subprocess
+import json
+
+class ChargingSession:
+    
+    def __init__(self):
+        self.test_case_json = "test_case.json"
+        self.json_tc = "Test Case"
+        self.range = "Range Output"
+        self.input = "Input Array"
+        self.root_dir = subprocess.Popen(['git', 'rev-parse', '--show-toplevel'],
+                                    stdout=subprocess.PIPE).communicate()[0].rstrip().decode('utf-8')
+    
+
+    # Pure Function Inside Class
+    def get_json_path(self, json_file_name, root_dir):
+        for roots, _, files in os.walk(root_dir):
+            if json_file_name in files:
+                return os.path.join(roots, json_file_name)
+
+    def get_json_data(self, json_path):
+        with open(json_path, "r") as jsonread:
+            json_data = json.load(jsonread)
+        return json_data
+
+    def null_data_check(self, data):
+        if len(data) == 0:
+            return False
+        else:
+            return True
+    
+    def single_data_check(self, data):
+        global range_list
+        if len(data) == 1:
+            range_list.append(data)
+            return False
+        else:
+            return True
+    
+    def last_single_data_break(self, data):
+        global range_list
+        if len(data) == 1:
+            if max(range_list[-1]) != data[-1]:
+                range_list.append(data)
+            return True
+        return False
+
+    def multiple_data_check(self, data, nested_list, max_idx):
```

Comment:
See if you can simplify this function. Is it the simplest code that will pass your tests?
The role of `max_idx` isn't clear to me